api/api/admin/media_report.py

```diff
@@ -222,6 +222,110 @@ def queryset(self, request, qs):
+                value = params.pop(self.parameter_name)
+                self.used_parameters[self.parameter_name] = value
+
+            self.range_min = first.id if (first := MediaDecision.objects.first()) else 0
+            self.range_max = last.id if (last := MediaDecision.objects.last()) else 0
+
+        def has_output(self) -> bool:
+            """
+            Determine if the filter should be displayed. The filter is only
+            displayed if there is at least one decision for the media type.
+            """
+
+            return MediaDecision.objects.exists()
+
+        def choices(self, changelist):
+            """
+            Return two URL query strings, one for filtering by an arbitrary
+            decision ID and for clearing the filter.
+            """
+
+            return [
+                changelist.get_query_string({self.parameter_name: "VALUE"}),
+                changelist.get_query_string(remove=[self.parameter_name]),
+            ]
+
+        def expected_parameters(self):
+            return [self.parameter_name]
+
+        def queryset(self, request, queryset):
+            """
+            Filter the query set to only show UUIDs associated with the selected
+            media decision.
+            """
+
+            if self.value() is not None:
+                uuids = MediaDecisionThrough.objects.filter(
+                    decision_id=self.value()
+                ).values_list("media_obj_id")
+                queryset = queryset.filter(media_obj_id__in=uuids)
+            return queryset
+
+        def value(self):
+            """
+            Parse the value from the URL query string. Any non numerical value
+            will be treated as ``None``. Any value outside of the filter's range
+            will also be treated as ``None``.
+            """
+
+            try:
+                value = int(self.used_parameters.get(self.parameter_name))
+                if not self.range_min <= value <= self.range_max:
+                    return None
+                return value
+            except (TypeError, ValueError):
+                return None
+
+    return MediaDecisionFilter
+
+
+def get_single_bulk_moderation_filter(media_type: str):
+    class SingleBulkModerationFilter(admin.SimpleListFilter):
+        title = "media count"
+        parameter_name = "media_count"
+
+        def lookups(self, request, model_admin):
+            return [
+                ("single", "Single"),
+                ("bulk", "Bulk"),
+            ]
+
+        def queryset(self, request, queryset):
+            if self.value() not in {"single", "bulk"}:
+                return queryset
+
+            queryset = queryset.annotate(
+                media_count=Count(f"{media_type}decisionthrough")
+            )
+
+            if self.value() == "single":
+                return queryset.filter(media_count=1)
+            elif self.value() == "bulk":
+                return queryset.filter(media_count__gt=1)
```

Comment:
```suggestion
        def queryset(self, request, queryset):
    				filter_conditions = {
        			  "single": {"media_count": 1},
        			  "bulk": {"media_count__gt": 1},
    				}
    				
            if self.value() not in filter_conditions:
                return queryset

            queryset = queryset.annotate(
                media_count=Count(f"{media_type}decisionthrough")
            )

    				return queryset.filter(**filter_conditions[self.value()])
```

More of a question than a nit: would this be a bit more "pythonic"? It seems like a pattern I've come across in my limited Python experience. I'm genuinely curious though, this is less of a suggestion and more of a personal curiosity.

Reply:
I don't have a preference for either of the two, and yours seems much cleaner. So I updated it in 45837f0.